src/phoenix/server/app.py

```diff
@@ -104,6 +109,37 @@ async def get(self, request: Request) -> FileResponse:
         )
 
 
+T = TypeVar("T", contravariant=True)
+
+
+class SupportsPut(Protocol[T]):
+    def put(self, obj: T) -> None:
+        ...
+
+
+class SpanHandler(HTTPEndpoint):
+    queue: SupportsPut[pb.Span]
+
+    async def post(self, request: Request) -> Response:
+        try:
+            content_type = request.headers.get("content-type")
+            if content_type == "application/x-protobuf":
+                body = await request.body()
+                content_encoding = request.headers.get("content-encoding")
+                if content_encoding == "gzip":
+                    body = gzip.decompress(body)
+                pb_span = pb.Span()
+                pb_span.ParseFromString(body)
+            else:
+                span = json_to_span(await request.json())
+                assert isinstance(span, Span)
```

Comment:
It's strange to me that `json_to_span` can return anything and that a type check is needed here.